### PR TITLE
Engines Modularized

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#145 

#### What does this implement/fix? Explain your changes.
There were some necessary changes regarding engine modularization. Now different engines by same interface is accessible through `pyrgg.engines`. Also tests separated for different modules.

### CODE REVIEW

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Thanks for your efforts 🥇 
Please take a look at my comments 🔥

File: otherfile/pyrgg_profile.py [hunk 1]
```diff
@@ -2,20 +2,24 @@
 """Profile file."""
 from pyrgg import *
 import pyrgg.params
+import pyrgg.engines.pyrgg as pyrgg_engine
```
[the remark this answers is not in the document]
Comment (sadrasabouri) [reply otherfile/pyrgg_profile.py]: @sepandhaghighi I added it in a separate profiler for easier tracking.
The problem is that now the the CLI killed ERG profile. I guess it's memory shortage problem. I suspect a memory leakage or garbage collector problem. Should I decrees the number of vertices for the ERG profiler?

[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply otherfile/pyrgg_profile.py]: @sadrasabouri Hi
I have resolved the issue of memory shortage (https://github.com/sepandhaghighi/pyrgg/pull/160/commits/cdb8db0eed8c7b93cf48caecefae07f2d9ecc660 and  https://github.com/sepandhaghighi/pyrgg/pull/160/commits/88436c399c1feeed84b2c26e6e7570391008b92c). Please review the entire PR again. If everything looks good, then ping me to finalize the PR review and merge. Thank you.

**Comment by sadrasabouri** [comment]
@sepandhaghighi It's ready for your review Sepand.

File: pyrgg/params.py [hunk 2]
```diff
@@ -56,7 +58,19 @@
...
+}
+
+ENGINE_MENU_INV = {v: k for k, v in ENGINE_MENU.items()}
+
+ERG_ENINGE_PARAMS = {
+    1: ["vertices", "- Vertices Number (>=0) : "],
+    2: ["probability", "- Probability (0 <= p <= 1) : "],
+}
+
+PYRGG_ENINGE_PARAMS = {
```
Comment (sepandhaghighi) [inline pyrgg/params.py:73]: `ENINGE` or `ENGINE`?

File: pyrgg/params.py [hunk 3]
```diff
@@ -56,7 +58,19 @@
...
 
-PYRGG_ENGINE_PARAMS = {
+ENGINE_MENU = {
+    1: "pyrgg",
+    2: "erg",
+}
+
+ENGINE_MENU_INV = {v: k for k, v in ENGINE_MENU.items()}
+
+ERG_ENINGE_PARAMS = {
```
Comment (sepandhaghighi) [inline pyrgg/params.py:68]: `ENINGE` or `ENGINE`?

File: test/functions_test.py [hunk 4]
```diff
@@ -186,8 +124,16 @@
 ...                return input_func_dict[item1]
 ...            else:
 ...                return input_func_dict["error"]
-...    for index in pyrgg.params.PYRGG_ENGINE_PARAMS:
-...        item1, item2 = pyrgg.params.PYRGG_ENGINE_PARAMS[index]
+...    for index in pyrgg.params.PYRGG_ENINGE_PARAMS:
```
Comment (sepandhaghighi) [inline test/functions_test.py:127]: `ENINGE` or `ENGINE`?

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Please take a look at my comments. I noticed some variable names that seem to be misspelled.

File: test/functions_test.py (hunk 4 above)
Comment (sadrasabouri) [reply test/functions_test.py]: fixed in [4728ff1](https://github.com/sepandhaghighi/pyrgg/pull/160/commits/4728ff19de9d07c9b01b1d9c80eba786bd80e685).

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**